ui/src/main/webapp/wizards/sources/validation.js

```diff
@@ -0,0 +1,35 @@
+export const isEmpty = (string) => {
+  return !string
+}
+
+export const isBlank = (string) => {
+  return !string || !string.trim()
+}
+
+export const nextShouldBeDisabled = ({ configs, discoveryType }) => {
+  // checks that username & password are either both filled out or both empty (because it's optional)
+  if (isBlank(configs.sourceUserName) !== isEmpty(configs.sourceUserPassword)) {
+    return true
+  }
+
+  // hostname & port discovery checks
+  if (discoveryType === 'hostnamePort') {
+    if (isBlank(configs.sourceHostName) || configs.sourcePort < 0 || configs.sourcePort > 65535) {
+      return true
+    }
+  }
+
+  // url discovery checks
+  if (discoveryType === 'url') {
+    if (isBlank(configs.endpointUrl)) {
```

Comment:
I think we can use [`validator.isURL`](https://github.com/chriso/validator.js#validators) here to improve client side validation.